### PR TITLE
automate OLM test case OCP-24074: check OLM resources version

### PR DESCRIPTION
**Pull request by jianzhangbjz**
Since some changes on 4.2, but not 4.1. So I think we can add automation test cases here. @bparees @njhale Could you help have a review? Thanks!
@stevekuznetsov @vikaslaad For 4.1.z automation, the branch `release-4.1` is the right place, right? I guess 4.1.z Errata conformance test will use this branch, right?

### CODE REVIEW

File: test/extended/operators/olm.go [hunk 1]
```diff
@@ -37,6 +38,52 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
...
+		"packagemanifests":       "v1",
+		"catalogsources":         "v1alpha1",
+		"subscriptions":          "v1alpha1",
+		"installplans":           "v1alpha1",
+		"clusterserviceversions": "v1alpha1",
+	}
+
+	for k, v := range versionMap {
+		g.It("check API version: "+k, func() {
+			output, err := oc.AsAdmin().Run("get").Args(k, "-n", "openshift-operator-lifecycle-manager", "-o=jsonpath={.items[0].apiVersion}").Output()
```
Comment (njhale) [inline test/extended/operators/olm.go]: Even though the earlier test does this, I don't think we can continue to expect an instance of each kind to exist in the `openshift-operator-lifecycle-manager` namespace.

File: test/extended/operators/olm.go [hunk 2]
```diff
@@ -37,6 +38,52 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
...
+		})
+		g.It("check OLM's CRDs' version: "+k, func() {
+			var output string
+			var err error
+
+			if k == "packagemanifests" {
+				// packagemanifests is a resource created by Aggregation layer, not CRD.
+				output, err = oc.AsAdmin().Run("get").Args("apiservice", "v1.packages.operators.coreos.com", "-o=jsonpath={.spec.version}").Output()
+			} else {
+				output, err = oc.AsAdmin().Run("get").Args("crd", fmt.Sprintf("%s.operators.coreos.com", k), "-o=jsonpath={.spec.version}").Output()
```
Comment (njhale) [inline test/extended/operators/olm.go]: (see https://github.com/openshift/origin/pull/23224#discussion_r300717900)

**Comment by stevekuznetsov** [comment]
Yep, this is the appropriate branch to make this against.

File: test/extended/operators/olm.go [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 import (
-	"reflect"
-
+	"fmt"
 	g "github.com/onsi/ginkgo"
 	o "github.com/onsi/gomega"
 	yaml "gopkg.in/yaml.v2"
 	e2e "k8s.io/kubernetes/test/e2e/framework"
+	"reflect"
+	"strings"
```
Comment (eparis) [inline test/extended/operators/olm.go]: These 2 stdlib imports belong up with "fmt".

**Comment by eparis** [comment]
Where is the master/4.2 PR? I'm trying to connect the dots.

**Comment by jianzhangbjz** [comment]
@eparis Here is that PR for master: #23224. But, it has been closed since PR #23331 covered this feature.

**Comment by eparis** [comment]
if this was covered by 23331 should this be closed? should we backport 23331 instead of 23224 ?

**Comment by jianzhangbjz** [comment]
@eparis Sorry for the late to reply. Yes, I think so.

**Pull request closed without merging**